File: javascript/secret-handshake/secret-handshake.js
//
// This is only a SKELETON file for the 'Secret Handshake' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const commands = (command) => {
  let handshake = [];
  if (command & 1) handshake.push("wink");
  if (command & 2) handshake.push("double blink");
  if (command & 4) handshake.push("close your eyes");
  if (command & 8) handshake.push("jump");
  if (command >= 16) handshake.reverse();

  return handshake;
};

File: dart/secret-handshake/lib/secret_handshake.dart
class SecretHandshake {
  List<String> commands( int i) {
    List<String> result = List.empty(growable: true);
    if (i & 1 == 1) result.add('wink');
    if (i & 2 == 2) result.add('double blink');
    if (i & 4 == 4) result.add('close your eyes');
    if (i & 8 == 8) result.add('jump');

    return (i & 16 == 16) ? result.reversed.toList() : result;
  }
}
